omod/src/main/java/org/openmrs/module/idgen/rest/resource/AutoGenerationOptionResource.java

```diff
@@ -0,0 +1,246 @@
+			description = new DelegatingResourceDescription();
+			description.addProperty("uuid");
+			description.addProperty("identifierType");
+			description.addProperty("location");
+			description.addProperty("source");
+			description.addProperty("manualEntryEnabled");
+			description.addProperty("automaticGenerationEnabled");
+			description.addSelfLink();
+			description.addLink("full", ".?v=" + RestConstants.REPRESENTATION_FULL);
+		} else if (rep instanceof FullRepresentation) {
+			description = new DelegatingResourceDescription();
+			description.addProperty("uuid");
+			description.addProperty("identifierType");
+			description.addProperty("location");
+			description.addProperty("source");
+			description.addProperty("manualEntryEnabled");
+			description.addProperty("automaticGenerationEnabled");
+			description.addSelfLink();
+		}
+		return description;
+	}
+	
+	@PropertyGetter("location")
+	public String getLocation(AutoGenerationOption autoGenerationOption) {
+		return autoGenerationOption.getLocation().getUuid();
+	}
+	
+	@PropertyGetter("identifierType")
+	public String getIdentifierType(AutoGenerationOption autoGenerationOption) {
+		return autoGenerationOption.getIdentifierType().getUuid();
+	}
+	
+	@PropertyGetter("source")
+	public String getSource(AutoGenerationOption autoGenerationOption) {
+		return autoGenerationOption.getSource().getUuid();
+	}
+	
+	@PropertyGetter("uuid")
+	public String getUuid(AutoGenerationOption autoGenerationOption) {
+		return autoGenerationOption.getUuid();
+	}
+	
+	@PropertyGetter("display")
+	public String getDisplayString(AutoGenerationOption autoGenerationOption) {
+		return autoGenerationOption.getIdentifierType() + " - " + autoGenerationOption.getSource() + " - "
+		        + autoGenerationOption.getLocation();
+	}
+	
+	@Override
+	public DelegatingResourceDescription getCreatableProperties() {
+		DelegatingResourceDescription description = new DelegatingResourceDescription();
+		description.addRequiredProperty("identifierType");
+		description.addProperty("location");
+		description.addRequiredProperty("source");
+		description.addProperty("manualEntryEnabled");
+		description.addProperty("automaticGenerationEnabled");
+		return description;
+	}
+	
+	@Override
+	public DelegatingResourceDescription getUpdatableProperties() {
+		DelegatingResourceDescription description = getCreatableProperties();
+		description.removeProperty("identifierType");
+		return description;
+	}
+	
+	@Override
+	public SimpleObject create(SimpleObject postBody, RequestContext context) throws ResponseException {
+		Object isManualEntryEnabled = (postBody.get("manualEntryEnabled"));
+		Object isAutomaticallyEnabled = postBody.get("automaticGenerationEnabled");
+		Object sourceUuid = postBody.get("source");
+		Object locationUuid = postBody.get("location");
+		Object identifierTypeUuid = postBody.get("identifierType");
+		String manualEntryEnabled = ((isManualEntryEnabled != null)
```

Comment:
@kwahalf can you deal with this exactly like how @adamsdenniskariuki dealt with it?
I mean the checks for nulls and assignment.